config/initializers/apipie.rb

```diff
@@ -173,3 +173,29 @@ def expected_type
     :any_type
   end
 end
+
+class DelayedEnumValidator < Apipie::Validator::BaseValidator
+  def initialize(param_description, block)
+    super(param_description)
+    @block = block
+  end
+
+  def self.build(param_description, argument, options, block)
+    if argument == :delayed_enum
+      new(param_description, options[:of])
+    end
+  end
+
+  def values
+    @block&.call || []
+  end
+
+  def validate(value)
+    values.include?(value)
+  end
+
+  def description
+    enum = values.map { |value| format_description_value(value) }.join(', ')
+    "Must be one of: #{enum}."
```

Comment:
Should we make this translatable?

Reply:
I don't see any of those being translatable neither in apipie-rails nor above.

If we want to make this translatable, we should probably do it for every validator.